pkg/crossplane/application.go

```diff
@@ -0,0 +1,79 @@
+package crossplane
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var (
+	fieldsAppDesiredRes          = append(fieldsStatus, "desiredResources")
+	fieldsAppSubmittedRes        = append(fieldsStatus, "submittedResources")
+	fieldsAppResourceSelector    = append(fieldsSpec, "resourceSelector")
+	fieldsAppResourceMatchLabels = append(fieldsAppResourceSelector, "matchLabels")
+
+	stateAppSubmitted = "Submitted"
+)
+
+type Application struct {
+	u *unstructured.Unstructured
+}
+
+func NewApplication(u *unstructured.Unstructured) *Application {
+	return &Application{u: u}
+}
+
+func (o *Application) GetStatus() string {
+	return getNestedString(o.u.Object, fieldsStatusState...)
+}
+
+func (o *Application) GetAge() string {
+	return GetAge(o.u)
+}
+
+func (o *Application) GetObjectDetails() ObjectDetails {
+	u := o.u
+	if u == nil {
+		return ObjectDetails{}
+	}
+	return getObjectDetails(o.u)
+}
+
+func (o *Application) IsReady() bool {
+	return o.GetStatus() == stateAppSubmitted
+}
+
+func (o *Application) GetRelated(filterByLabel func(metav1.GroupVersionKind, string, string) ([]unstructured.Unstructured, error)) ([]*unstructured.Unstructured, error) {
```

Comment:
Something to keep in mind for the future is that it'd be great if we could add support for more types without having to recompile the `trace` binary. We don't have to have a solution now, but it's something to think about.

Reply:
Yes, a good point.
This is something that I also considered in the context of making this tool as a "general purpose" tracing tool as you mentioned in the https://github.com/crossplaneio/crossplane/issues/744. This could be possible with some sort of configuration file which describes the types and their relations with other types.